File: mobile_app/app_ios/Objects/src/B4iProject/LS_frmprincipal.m

#import "iCore.h"
@interface LS_frmprincipal:NSObject
@end

@implementation LS_frmprincipal

- (void)LS_general:(B4ILayoutData*)views :(int)width :(int)height{
[B4ILayoutBuilder setScaleRate:0.3];
[B4ILayoutBuilder scaleAll:views :width :height];
[views get:@"btnmasvendido"].Top = (int)((2 / 100.0 * height));
[views get:@"btnmasvendido"].Left = (int)((3 / 100.0 * width));
[views get:@"btnmasvendido"].Width = (int)((46 / 100.0 * width));
[views get:@"btnmasvendido"].Height = (int)((7 / 100.0 * height));
[views get:@"btnbuscarproducto"].Left = (int)(([views get:@"btnmasvendido"].Left + [views get:@"btnmasvendido"].Width)+(2 / 100.0 * width));
[views get:@"btnbuscarproducto"].Width = (int)(([views get:@"btnmasvendido"].Width));
[views get:@"btnbuscarproducto"].Height = (int)(([views get:@"btnmasvendido"].Height));
[views get:@"btnbuscarproducto"].Top = (int)(([views get:@"btnmasvendido"].Top));
[views get:@"lblcategoria"].Top = (int)(([views get:@"btnmasvendido"].Top + [views get:@"btnmasvendido"].Height)+(1 / 100.0 * height));
[views get:@"lblcategoria"].Left = (int)(([views get:@"btnmasvendido"].Left));
[views get:@"lblcategoria"].Height = (int)((5 / 100.0 * height));
[views get:@"lblcategoria"].Width = (int)((90 / 100.0 * width));
[views get:@"panelcategoria"].Left = (int)(([views get:@"lblcategoria"].Left));
[views get:@"panelcategoria"].Top = (int)(([views get:@"lblcategoria"].Top + [views get:@"lblcategoria"].Height));
[views get:@"panelcategoria"].Width = (int)((100 / 100.0 * width)-([views get:@"viewcategoria"].Left)*2.0);
[views get:@"panelcategoria"].Height = (int)((77 / 100.0 * height));
[views get:@"viewcategoria"].Left = (int)(0.0);
[views get:@"viewcategoria"].Top = (int)(0.0);
[views get:@"viewcategoria"].Width = (int)(([views get:@"panelcategoria"].Width));
[views get:@"viewcategoria"].Height = (int)(([views get:@"panelcategoria"].Height));
[views get:@"btnabierto"].Top = (int)(([views get:@"panelcategoria"].Top + [views get:@"panelcategoria"].Height)+(1.5 / 100.0 * height));
[views get:@"btnabierto"].Width = (int)((35 / 100.0 * width));
[views get:@"btnabierto"].Left = (int)(([views get:@"panelcategoria"].Left));
[views get:@"btnabierto"].Height = (int)((5.5 / 100.0 * height));
[views get:@"lbllema"].Left = (int)(([views get:@"btnabierto"].Left + [views get:@"btnabierto"].Width)+(2 / 100.0 * width));
[views get:@"lbllema"].Top = (int)(([views get:@"panelcategoria"].Top + [views get:@"panelcategoria"].Height));
[views get:@"lbllema"].Height = (int)((100 / 100.0 * height)-([views get:@"lbllema"].Top));
[views get:@"lbllema"].Width = (int)((100 / 100.0 * width)-([views get:@"lbllema"].Left));
[views get:@"panelmodalproducto"].Width = (int)((100 / 100.0 * width));
[views get:@"panelmodalproducto"].Left = (int)((100 / 100.0 * width));
[views get:@"panelmodalproducto"].Top = (int)(0.0);
[views get:@"panelmodalproducto"].Height = (int)((100 / 100.0 * height) - (0.0));
[views get:@"panel1"].Top = (int)((9 / 100.0 * height));
[views get:@"panel1"].Left = (int)((5 / 100.0 * width));
[views get:@"panel1"].Height = (int)((9 / 100.0 * height));
[views get:@"panel1"].Width = (int)((100 / 100.0 * width)-([views get:@"panel1"].Left)*2.0);
[views get:@"txtbuscarproducto"].Left = (int)((1 / 100.0 * width));
[views get:@"txtbuscarproducto"].Top = (int)((0.9 / 100.0 * height));
[views get:@"txtbuscarproducto"].Width = (int)((72 / 100.0 * width));
[views get:@"txtbuscarproducto"].Height = (int)(([views get:@"panel1"].Height)-([views get:@"txtbuscarproducto"].Top)*2.0);
[views get:@"btnproductobuscarmodal"].Left = (int)(([views get:@"txtbuscarproducto"].Left + [views get:@"txtbuscarproducto"].Width)+(1 / 100.0 * width));
[views get:@"btnproductobuscarmodal"].Height = (int)(([views get:@"txtbuscarproducto"].Height));
[views get:@"btnproductobuscarmodal"].Width = (int)((15 / 100.0 * width));
[views get:@"btnproductobuscarmodal"].Top = (int)(([views get:@"txtbuscarproducto"].Top));
[views get:@"panel2"].Left = (int)(([views get:@"panel1"].Left));
[views get:@"panel2"].Top = (int)(([views get:@"panel1"].Top + [views get:@"panel1"].Height)+(2 / 100.0 * height));
//BA.debugLineNum = 62;BA.debugLine="Panel2.Width = Panel1.Width"[frmPrincipal/General script]
[views get:@"panel2"].Width = (int)(([views get:@"panel1"].Width));
//BA.debugLineNum = 63;BA.debugLine="Panel2.Height = 75%y"[frmPrincipal/General script]
[views get:@"panel2"].Height = (int)((75 / 100.0 * height));
//BA.debugLineNum = 65;BA.debugLine="PanelProductosModalView.Left = 2%x"[frmPrincipal/General script]
[views get:@"panelproductosmodalview"].Left = (int)((2 / 100.0 * width));
//BA.debugLineNum = 66;BA.debugLine="PanelProductosModalView.Top = 1%y"[frmPrincipal/General script]
[views get:@"panelproductosmodalview"].Top = (int)((1 / 100.0 * height));
//BA.debugLineNum = 67;BA.debugLine="PanelProductosModalView.Width = Panel2.Width - PanelProductosModalView.Left*2"[frmPrincipal/General script]
[views get:@"panelproductosmodalview"].Width = (int)(([views get:@"panel2"].Width)-([views get:@"panelproductosmodalview"].Left)*2.0);
//BA.debugLineNum = 68;BA.debugLine="PanelProductosModalView.Height = Panel2.Height - 11%y"[frmPrincipal/General script]
[views get:@"panelproductosmodalview"].Height = (int)(([views get:@"panel2"].Height)-(11 / 100.0 * height));
//BA.debugLineNum = 70;BA.debugLine="btnProductoVolver.Top = PanelProductosModalView.Bottom + 1%y"[frmPrincipal/General script]
[views get:@"btnproductovolver"].Top = (int)(([views get:@"panelproductosmodalview"].Top + [views get:@"panelproductosmodalview"].Height)+(1 / 100.0 * height));
//BA.debugLineNum = 71;BA.debugLine="btnProductoVolver.Left = PanelProductosModalView.Left"[frmPrincipal/General script]
[views get:@"btnproductovolver"].Left = (int)(([views get:@"panelproductosmodalview"].Left));
//BA.debugLineNum = 72;BA.debugLine="btnProductoVolver.Width = PanelProductosModalView.Width"[frmPrincipal/General script]
[views get:@"btnproductovolver"].Width = (int)(([views get:@"panelproductosmodalview"].Width));
//BA.debugLineNum = 73;BA.debugLine="btnProductoVolver.Height = 7%y"[frmPrincipal/General script]
[views get:@"btnproductovolver"].Height = (int)((7 / 100.0 * height));
//BA.debugLineNum = 74;BA.debugLine="Panel2.Height = btnProductoVolver.Bottom +1%y"[frmPrincipal/General script]
[views get:@"panel2"].Height = (int)(([views get:@"btnproductovolver"].Top + [views get:@"btnproductovolver"].Height)+(1 / 100.0 * height));

}
@end

File: mobile_app/app/Objects/src/com/menube/app/designerscripts/LS_frmprincipal.java
package com.menube.app.designerscripts;
import anywheresoftware.b4a.objects.TextViewWrapper;
import anywheresoftware.b4a.objects.ImageViewWrapper;
import anywheresoftware.b4a.BA;


public class LS_frmprincipal{

public static void LS_general(java.util.LinkedHashMap<String, anywheresoftware.b4a.keywords.LayoutBuilder.ViewWrapperAndAnchor> views, int width, int height, float scale) {
anywheresoftware.b4a.keywords.LayoutBuilder.setScaleRate(0.3);
anywheresoftware.b4a.keywords.LayoutBuilder.scaleAll(views);
views.get("panelmenu").vw.setLeft((int)(0d));
views.get("panelmenu").vw.setWidth((int)((100d / 100 * width) - (0d)));
views.get("panelmenu").vw.setTop((int)(0d));
views.get("panelmenu").vw.setHeight((int)((100d / 100 * height) - (0d)));
views.get("paneltitulo").vw.setLeft((int)((0d / 100 * width)));
views.get("paneltitulo").vw.setWidth((int)((100d / 100 * width) - ((0d / 100 * width))));
views.get("paneltitulo").vw.setTop((int)((0d / 100 * height)));
views.get("paneltitulo").vw.setHeight((int)((8d / 100 * height) - ((0d / 100 * height))));
views.get("btnprincipalback").vw.setLeft((int)(0d));
views.get("btnprincipalback").vw.setTop((int)(0d));
views.get("btnprincipalback").vw.setWidth((int)((15d / 100 * width)));
views.get("btnprincipalback").vw.setHeight((int)((views.get("paneltitulo").vw.getHeight())));
views.get("btnempresainfo").vw.setTop((int)(0d));
views.get("btnempresainfo").vw.setLeft((int)((views.get("btnprincipalback").vw.getLeft() + views.get("btnprincipalback").vw.getWidth())));
views.get("btnempresainfo").vw.setHeight((int)((views.get("btnprincipalback").vw.getHeight())));
views.get("btnempresainfo").vw.setWidth((int)((100d / 100 * width)-(views.get("btnempresainfo").vw.getLeft())*2d));
views.get("btnorden").vw.setLeft((int)((views.get("btnempresainfo").vw.getLeft() + views.get("btnempresainfo").vw.getWidth())));
views.get("btnorden").vw.setWidth((int)((views.get("btnprincipalback").vw.getWidth())));
views.get("btnorden").vw.setHeight((int)((views.get("btnprincipalback").vw.getHeight())));
views.get("btnorden").vw.setTop((int)((views.get("btnprincipalback").vw.getTop())));
views.get("lblcantidadproductos").vw.setHeight((int)((23d * scale)));
views.get("lblcantidadproductos").vw.setWidth((int)((views.get("lblcantidadproductos").vw.getHeight())));
views.get("lblcantidadproductos").vw.setLeft((int)((views.get("btnorden").vw.getLeft() + views.get("btnorden").vw.getWidth())-(1d / 100 * width) - (views.get("lblcantidadproductos").vw.getWidth())));
views.get("lblcantidadproductos").vw.setTop((int)((views.get("btnorden").vw.getTop())+(0.5d / 100 * height)));
views.get("panel1").vw.setTop((int)((views.get("paneltitulo").vw.getTop() + views.get("paneltitulo").vw.getHeight())+(1d / 100 * height)));
views.get("panel1").vw.setLeft((int)((3d / 100 * width)));
views.get("panel1").vw.setWidth((int)((46d / 100 * width)));
views.get("panel1").vw.setHeight((int)((7d / 100 * height)));
views.get("btnmasvendido").vw.setLeft((int)(0d));
views.get("btnmasvendido").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("btnmasvendido").vw.setHeight((int)((views.get("panel1").vw.getHeight())));
views.get("btnmasvendido").vw.setTop((int)(0d));
views.get("panel2").vw.setLeft((int)((views.get("panel1").vw.getLeft() + views.get("panel1").vw.getWidth())+(2d / 100 * width)));
views.get("panel2").vw.setWidth((int)((views.get("panel1").vw.getWidth())));
views.get("panel2").vw.setHeight((int)((views.get("panel1").vw.getHeight())));
views.get("panel2").vw.setTop((int)((views.get("panel1").vw.getTop())));
views.get("btnbuscarproducto").vw.setLeft((int)(0d));
views.get("btnbuscarproducto").vw.setWidth((int)((views.get("panel2").vw.getWidth())));
views.get("btnbuscarproducto").vw.setHeight((int)((views.get("panel2").vw.getHeight())));
views.get("btnbuscarproducto").vw.setTop((int)(0d));
views.get("lblcategoria").vw.setTop((int)((views.get("panel1").vw.getTop() + views.get("panel1").vw.getHeight())+(1d / 100 * height)));
views.get("lblcategoria").vw.setLeft((int)((views.get("panel1").vw.getLeft())));
views.get("lblcategoria").vw.setHeight((int)((5d / 100 * height)));
views.get("lblcategoria").vw.setWidth((int)((90d / 100 * width)));
views.get("panelcategoria").vw.setLeft((int)((views.get("lblcategoria").vw.getLeft())));
views.get("panelcategoria").vw.setTop((int)((views.get("lblcategoria").vw.getTop() + views.get("lblcategoria").vw.getHeight())));
views.get("panelcategoria").vw.setWidth((int)((100d / 100 * width)-(views.get("panelcategoria").vw.getLeft())*2d));
views.get("panelcategoria").vw.setHeight((int)((69d / 100 * height)));
views.get("viewcategoria").vw.setLeft((int)((1d / 100 * width)));
views.get("viewcategoria").vw.setHeight((int)((views.get("panelcategoria").vw.getHeight())));
views.get("viewcategoria").vw.setWidth((int)((views.get("panelcategoria").vw.getWidth())-(views.get("viewcategoria").vw.getLeft())*2d));
views.get("viewcategoria").vw.setTop((int)(0d));
views.get("btnabierto").vw.setTop((int)((views.get("panelcategoria").vw.getTop() + views.get("panelcategoria").vw.getHeight())+(1.5d / 100 * height)));
//BA.debugLineNum = 67;BA.debugLine="btnAbierto.Width = 35%x"[frmPrincipal/General script]
views.get("btnabierto").vw.setWidth((int)((35d / 100 * width)));
//BA.debugLineNum = 68;BA.debugLine="btnAbierto.Left = PanelCategoria.Left"[frmPrincipal/General script]
views.get("btnabierto").vw.setLeft((int)((views.get("panelcategoria").vw.getLeft())));
//BA.debugLineNum = 69;BA.debugLine="btnAbierto.Height = 5.5%y"[frmPrincipal/General script]
views.get("btnabierto").vw.setHeight((int)((5.5d / 100 * height)));
//BA.debugLineNum = 71;BA.debugLine="lblLema.Left = btnAbierto.Right + 2%x"[frmPrincipal/General script]
views.get("lbllema").vw.setLeft((int)((views.get("btnabierto").vw.getLeft() + views.get("btnabierto").vw.getWidth())+(2d / 100 * width)));
//BA.debugLineNum = 72;BA.debugLine="lblLema.Top = PanelCategoria.Bottom"[frmPrincipal/General script]
views.get("lbllema").vw.setTop((int)((views.get("panelcategoria").vw.getTop() + views.get("panelcategoria").vw.getHeight())));
//BA.debugLineNum = 73;BA.debugLine="lblLema.Height = 100%y - lblLema.Top"[frmPrincipal/General script]
views.get("lbllema").vw.setHeight((int)((100d / 100 * height)-(views.get("lbllema").vw.getTop())));
//BA.debugLineNum = 74;BA.debugLine="lblLema.Width = 100%x - lblLema.Left"[frmPrincipal/General script]
views.get("lbllema").vw.setWidth((int)((100d / 100 * width)-(views.get("lbllema").vw.getLeft())));

}
}